### ISSUE TITLE
Need examples or guide

### ISSUE THREAD

**Issue by David-OConnor** [bluss/petgraph#258]
The API docs are all I could find in the way of documentation. Needs examples and/or instructions. There are a few trivial examples in the API docs under `Graph`, but that's all I could find.

**Comment by ultrasaurus** [comment]
I agree that a guide would be great.  For starters, examples that cover use cases not included in the docs could be helpful.

**Comment by jmcomets** [comment]
Another issue raised by @m4b:

> Some more examples would be helpful, especially on how to use certain iterators, e.g.: https://docs.rs/petgraph/0.4.9/petgraph/graph/struct.Graph.html#method.node_indices
>
> would be nice to show stuff like.

```rust
for id in cfg.node_indices() {
    if let Some(&thing) = cfg.node_weight(id) {
```

I also personally think we need clearer examples showing which graphs data structures should be used for some typical use cases (mentioned in #245)

**Comment by bluss** [comment]
Those that can think of it, please supply questions that the docs could answer with examples. It's great if it is a set of common tasks (broken down to distinct small tasks).

**Comment by timthelion** [comment]
I'm working on a review here: https://thobbs.cz/rust-play/petgraph_review.html
Perhaps this could be cannibalized for such a purpose?

**Comment by timthelion** [comment]
I just added another guide, the first one was tutorials, this one covers internals: https://timothy.hobbs.cz/rust-play/petgraph-internals.html

**Comment by RaoulLuque** [comment]
I think this might be improved/resolved by https://github.com/petgraph/petgraph/pull/807. I would nominate this as to be closed, but if you have further suggestions, what examples could be useful, they would be appreciated :)

There is an argument to be made for writing a more long-form/thorough guide/tutorial, but I think we could open a new issue for that to discuss such an endeavor :)
